code/modules/admin/verbs/playsound.dm

```diff
@@ -57,10 +70,16 @@ GLOBAL_LIST_EMPTY(sounds_cache)
 	var/list/sounds = file2list("sound/serversound_list.txt")
 	sounds += GLOB.sounds_cache
 
-	var/melody = input("Select a sound from the server to play", "Server sound list") as null|anything in sounds
+	var/sound/melody = input("Select a sound from the server to play", "Server sound list") as null|anything in sounds
 	if(!melody)	return
 
-	play_sound(melody)
+	log_admin("[key_name(src)] played server sound [melody]")
+	message_admins("[key_name_admin(src)] played server sound [melody]")
+
+	for(var/mob/M in GLOB.player_list)
+		if(prefs.sound & SOUND_MIDI)
+			SEND_SOUND(src, sound(melody, volume = 85 * prefs.get_channel_volume(CHANNEL_ADMIN), channel = CHANNEL_ADMIN))
```

Comment:
You do realise this just checks the admins client and sends it to them *hundreds* of times right?


You dont read from `M` in this loop.

Please actually test your code.

Reply:
Sorry, this is fixed now.